src/misc/HtmlSanitizer.ts

```diff
@@ -341,15 +340,26 @@ export class HtmlSanitizer {
 		//     rgba(0, 0, 255, 0.5)
 		//   ), url("catfront.png");
 		// ```
-		// in this case background-image can have multiple values but it's safe to just block the whole thing
+		// in this case background-image can have multiple values, and only the url would be replaced
 		//
 		// some examples where it can be inside a single <image> value:
 		//
 		// cross-fade(20% url(twenty.png), url(eighty.png))
 		// image-set('test.jpg' 1x, 'test-2x.jpg' 2x)
-		if (value.includes("url(") && !value.match(/^url\(["']?data:/)) {
-			this.externalContent++
-			;(htmlNode.style as any)[styleAttributeName] = `url("${PREVENT_EXTERNAL_IMAGE_LOADING_ICON}")`
+		if (value.includes("url(")) {
+			const elements = value.split("),")
+			let finalElements = ""
+			for (const index in elements) {
```

Comment:
needing index is probably the only case where I would recommend `foreach()`. `for in` is tricky to use right and you also needed the casts below, also `for in` always looks like a bug, especially on arrays.

I have few alternatives that I want to suggest but see for yourself:

 * if there's some `url()` that is not `data:` URL, just replace the whole thing, like before. This could be quite simple (e.g. with regex)
 * or,  keep doing what you are doing but use regex for extracting the values
 * or, keep doing what you are doing but split by `, `, so that we don't have to think of parens

Reply:
I wanted to do bullet point one (and would honestly prefer it), but we discussed it and I understood that we want to preserve data URL, so if there's both (see below example) you have to extract the non-data urls. Is my understanding correct?

`<p style="background-image: url('data:,'), url('https://www.google.com/images/searchbox/desktop_searchbox_sprites318_hr.webp')"></p>`

Reply:
I think we only want to preserve them when they are the only value (like our signature), I don't think we have to care about it if it's a complex `background-image` so I think option 1 is good still.